source/slang/slang-check-decl.cpp

```diff
@@ -10057,6 +10276,8 @@ void SemanticsDeclBasesVisitor::visitExtensionDecl(ExtensionDecl* decl)
 
         _validateCrossModuleInheritance(decl, inheritanceDecl);
     }
+
+    validateDynUsage(this, getSink(), getOptionSet(), decl);
```

Comment:
I think if you move the related checks to `visitInheritanceDecl` you shouldn't need to call `validateDynUsage` here.